spec/formatting.md

```diff
@@ -505,14 +520,15 @@ _Formatting_ is a mostly implementation-defined process,
 as it depends on the implementation's shape for resolved values
 and the result type of the formatting.
 
-Formatting errors MAY be emitted during _formatting_,
-as formatting is not necessarily defined on every resolved value.
-A formatter MAY provide a value to use in such a case instead of a _fallback value_.
-
-_Formatting_ MAY produce formatted messages with the following data types,
-as well as any others:
+Resolved values cannot always be formatted by a given implementation.
+When such an error occurs during _formatting_, 
+an implementation SHOULD emit a _formatting error_ and produce a 
+_fallback value_ for the _placeholder_ that produced the error.
+A formatting function MAY substitute a value to use instead of a _fallback value_.
 
-- A single concatenated string.
+Implementations MAY represent the result of _formatting_ using the most
+appropriate data type or structure. Some examples of these include:
+- A single string concatenated from the resolved _pattern_ and its parts.
```

Comment:
```suggestion
- A single string concatenated from the parts of the resolved _pattern_.
```

I find it odd to refer separately to the _pattern_ and the _parts_. That implies that the _pattern_ includes something that's not a _part_ but does affect the result.